ls1intum/Artemis
# `Quiz exercises`: Allow to import exam quiz exercises with currently active test runs

### Checklist
#### General
#### Client

### Motivation and Context

Closes #5379. Allows users to import a quiz exercise that was used in an exam test run.

### Description

Added logic - it shouldn't be possible to edit a quiz used in an exam test run, **except** if we're actually importing it.

Added some tests to check that the alert service is/isn't called when editing a quiz used in an exam test run.

### Steps for Testing

Prerequisites:
- 1 Instructor

Similar to the steps described in #5379:

1. Create an exam with a quiz exercise
2. Make a test run on that exam using the quiz exercise
3. Create a new exam (could be in another course)
4. Import the quiz exercise used on the test run from the first exam
5. Check that you **do not** get the error that says "need to delete the test run"
6. Check that you **can** save the exercise without any errors. 

### Review Progress

#### Code Review
#### Manual Tests

### Test Coverage
| Class/File | Branch | Line |
|------------|-------:|-----:|
| quiz-exercise-detail.component.ts | 82% | 96% |
| quiz-exercise-validation.directive.ts | 91% | 98% |

### Screenshots
